### ISSUE TITLE
slow opening of database when huge data is avaliable 

### ISSUE THREAD

**Issue by VarmaTech** [tekartik/sembast.dart#191]
Below step take more than 50 seconds to open when huge data is available.( approx 3lakhs records are available in table)

await databaseFactoryIo.openDatabase(dbPath)

is there a way to speed up.

**Comment by alextekartik** [comment]
Can you confirm on which platform you are seeing this (web? mobile? android? linux?...) and information on the database (size in bytes and record count for each store - I'm not sure I got it right in your comment).
Also whether you are using the latests sembast (web? io? sembast_sqflite?).
Thanks.

Regarding performance, best is not to store binary data (images) and instead use external files and store reference (file path) in the database.

**Comment by VarmaTech** [comment]
Platform  : ANDROID
table : 15 nos
in which 
table 1 : 30000 records
table 2 : 290000 records 
table 3 : 52000 records 
remaining all have some random records between 500 to 4000 records .

No images or any media file or any binary data ,
 it's just data only.

**Comment by alextekartik** [comment]
Arg indeed that's a lot of data, I did not even think it could work with 300K records (I just found out what 3lakhs means sorry). I have not set hard limit but sembast all in memory design will significantly degrade in performance with so many records and I don't see any solution for that.

My suggestion would be to move to a **real** database such as SQLite on Android and IndexedDB on the web when dealing with such a database. Maybe that's not clear in sembast description, I will add a note on that.
